src/ctap/storage.rs

```diff
@@ -138,11 +138,14 @@ const PAGE_SIZE: usize = 0x100;
 #[cfg(not(feature = "ram_storage"))]
 const PAGE_SIZE: usize = 0x1000;
 
-const STORE_SIZE: usize = NUM_PAGES * PAGE_SIZE;
-
+// We have the following layout:
+// 0x00000-0x2ffff: Tock
+// 0x30000-0x3ffff: Padding
+// 0x40000-0xbffff: App
+// 0xc0000-0xfffff: Store
 #[cfg(not(any(test, feature = "ram_storage")))]
-#[link_section = ".app_state"]
-static STORE: [u8; STORE_SIZE] = [0xff; STORE_SIZE];
+const STORE_ADDR: usize = 0xC0000;
```

Comment:
Not a huge fan of having flash addresses that are hard-coded.
Maybe we should move this in the ldscript file and import the symbol instead?

Reply:
I'm also not a fan of hard-coded stuff but we can't have it in the linker script of the application because this should be in sync with the kernel. However, I moved the truth value to the kernel only. The application reads it from the kernel and is completely abstract over this information. The information is currently in 2 places:
- In the flash driver
- In the MPU configuration

When Tock supports persistent storage, we will be able to remove those 2 hard-coded values:
- The flash driver should ask the process whether the region is valid (like it was the case with writable flash regions)
- The kernel should configure the MPU properly when spawning a process (this was automatic with writable flash regions)

Reply:
It was briefly mentioned in the TockOS Slack that using the existing `nonvolatile_storage_driver.rs` capsule in combination with a [syscall filter](https://github.com/tock/tock/blob/3242eac532d999cd6fe15bb508bc84ba62ce4a5b/kernel/src/platform/mod.rs#L55) might allow us to implement our storage system that will only be available for the intended app.

Reply:
Those APIs don't support erasing a page. I did an analysis a [long time ago](https://github.com/tock/tock/pull/1457#issuecomment-551897059) and the situation did not change since then (except for this email thread that might end up somewhere in the next year or two).

We know we want to improve on the current state of this PR, but it can't be done until Tock supports persistent storage correctly. I don't see this happening in the next 6 months, because those are design decisions that take time. So if we believe we need this PR before that, we should submit, otherwise we can wait.